src/TypeScript.UnitTests/EslintBridgeClient/EslintBridgeProcessTests.cs

```diff
@@ -57,6 +58,54 @@ public void Start_FailsToFindNode_FileNotFoundException()
             act.Should().ThrowExactly<FileNotFoundException>().And.Message.Should().Contain("node.exe");
         }
 
+        [TestMethod]
+        public async Task Start_SecondCall_ProcessFailed_StartsNewProcess()
+        {
+            var fakeNodeExePath = CreateScriptThatPrintsPortNumber(123, isHanging: true);
+            var startupScriptPath = "dummy path";
+
+            var nodeLocator = SetupNodeLocator(fakeNodeExePath);
+            var testSubject = CreateTestSubject(startupScriptPath, nodeLocator: nodeLocator.Object);
+
+            await testSubject.Start();
+
+            var oldProcess = testSubject.Process;
+
+            oldProcess.HasExited.Should().BeFalse();
+            oldProcess.Kill();
+            await oldProcess.WaitForExitAsync();
+            oldProcess.HasExited.Should().BeTrue();
+
+            await testSubject.Start();
+
+            var newProcess = testSubject.Process;
+            newProcess.Should().NotBeSameAs(oldProcess);
+            newProcess.HasExited.Should().BeFalse();
+            newProcess.Kill();
```

Comment:
The cleanup code won't be executed if any of the assertions fail. It would be safer to always do the cleanup in a `finally` block.

I think only the `newProcess` has to be conditionally cleaned up in the `finally` block. There don't look to be any paths by which there could be an error that leaves the `oldProcess` running, although it would be safer to make sure.